MoltenVK/MoltenVK/Commands/MVKCommandResourceFactory.mm

```diff
@@ -551,6 +551,10 @@ static void getSwizzleString(char swizzleStr[4], VkComponentMapping vkMapping) {
 id<MTLComputePipelineState> MVKCommandResourceFactory::newCmdResolveColorImageMTLComputePipelineState(MVKFormatType type,
 																									  MVKVulkanAPIDeviceObject* owner,
 																									  bool isTextureArray) {
+#if MVK_TVOS
+    if (isTextureArray && !mvkOSVersionIsAtLeast(14))
+        isTextureArray = false;
+#endif
```

Comment:
At first, this looks wrong. If a 2D multisampled array texture were passed here in this case, we'd have to resolve each element of the array individually.

Then I realized that this shouldn't even be reached in that case, because the client **must** (in the normative, [RFC 2119][1] sense) not create a 2D multisampled array image if that image type isn't supported, or the ICD is permitted to make demons fly out of the client's nose. You could probably delete this code.

[1]: https://www.ietf.org/rfc/rfc2119.txt

Reply:
Removed, thanks!